src/main/java/com/overops/plugins/bamboo/TaskType.java

```diff
@@ -70,27 +67,43 @@ public TaskResult execute(@NotNull TaskContext context) throws TaskException {
 
         TaskResultBuilder resultBuilder = TaskResultBuilder.newBuilder(context);
 
+        logger.addBuildLogEntry(context.getConfigurationMap().get(Const.LINK));
+        Boolean displayLink = Boolean.parseBoolean(context.getConfigurationMap().get(Const.LINK));
         try {
-            logger.addBuildLogEntry("[" + Utils.getArtifactId() + " v" + Utils.getVersion() + "]");
-
-            boolean showAllEvents = Boolean.parseBoolean(context.getConfigurationMap().get(Const.SHOW_ALL_EVENTS));
-            boolean isDebug = Boolean.parseBoolean(context.getConfigurationMap().get(Const.DEBUG));
-            PrintStream printStream = isDebug ? new BambooPrintWriter(System.out, logger) : null;
-
-            QualityReport reportModel = overOpsService.runQualityReport(endPoint, apiKey, query, Requestor.BAMBOO, printStream, isDebug);
-
-            context.getBuildContext().getBuildResult().getCustomBuildData().put("overOpsReport", objectMapper.writeValueAsString(reportModel.getHtmlParts(showAllEvents)));
-            context.getBuildContext().getBuildResult().getCustomBuildData().put("isOverOpsStep", "true");
-
-            if (reportModel.getStatusCode() == ReportStatus.FAILED) {
-                if ((reportModel.getExceptionDetails() != null) && Boolean.parseBoolean(context.getConfigurationMap().get(Const.PASS_BUILD_ON_QR_EXCEPTION))) {
+            if (displayLink)
+            {
+                String appUrl = (String)globalSettings.get(Const.GLOBAL_APP_URL);
+                HtmlParts htmlParts = new HtmlParts(overOpsService.generateReportLinkHtml(appUrl, query), "");
+                context.getBuildContext().getBuildResult().getCustomBuildData().put("overOpsReport", objectMapper.writeValueAsString(htmlParts));
+                context.getBuildContext().getBuildResult().getCustomBuildData().put("isOverOpsStep", "true");
+                return resultBuilder.success().build();
+            } else
+            {
+                logger.addBuildLogEntry("[" + Utils.getArtifactId() + " v" + Utils.getVersion() + "]");
+
+                boolean showAllEvents = Boolean.parseBoolean(context.getConfigurationMap().get(Const.SHOW_ALL_EVENTS));
+                boolean isDebug = Boolean.parseBoolean(context.getConfigurationMap().get(Const.DEBUG));
+                PrintStream printStream = isDebug ? new BambooPrintWriter(System.out, logger) : null;
```

Comment:
I think this should move outside `if (displayLink)` as well.. we probably want to have debug logging regardless of the state of this variable to check for other settings such as API host, env ID, etc.

Reply:
Updated to show the major settings.